tensorbay/opendataset/CityscapesGTFine/loader.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+
+"""Dataloader of CityscapesGTFine dataset."""
+import json
+import os
+from glob import glob
+from typing import List
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.label import InstanceMask, LabeledPolygon, SemanticMask
+
+DATASET_NAME = "CityscapesGTFine"
+_SEGMENTS_INFO = {"test", "train", "val"}
+
+
+def CityscapesGTFine(path: str) -> Dataset:
+    """`CityscapesGTFine <https://www.cityscapes-dataset.com/>`_ dataset.
+
+    The file structure should be like::
+
+        <path>
+            leftImg8bit/
+                test/
+                    berlin/
+                        berlin_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                train/
+                    aachen/
+                        aachen_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                val/
+                    frankfurt/
+                        frankfurt_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                ...
+            gtFine/
+                test/
+                    berlin/
+                        berlin_000000_000019_gtFine_instanceIds.png
+                        berlin_000000_000019_gtFine_labelIds.png
+                        berlin_000000_000019_gtFine_polygons.json
+                        ...
+                    ...
+                train/
+                    aachen/
+                        aachen_000000_000019_gtFine_instanceIds.png
+                        aachen_000000_000019_gtFine_labelIds.png
+                        aachen_000000_000019_gtFine_polygons.json
+                        ...
+                    ...
+                val/
+                    frankfurt/
+                        frankfurt_000000_000019_gtFine_instanceIds.png
+                        frankfurt_000000_000019_gtFine_labelIds.png
+                        frankfurt_000000_000019_gtFine_polygons.json
+                        ...
+                    ...
+            ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENTS_INFO:
+        segment = dataset.create_segment(segment_name)
+        image_paths = glob(os.path.join(root_path, "leftImg8bit", "*", "*", "*.png"))
+        for image_path in image_paths:
+            city = image_path.split("_")[0]
+            stem = "_".join(image_path.split("_")[0:4])
```

Comment:
`stem` means the filename without extension. This is not a good place to use it.

Also split with `"_"` then use `"_"` to join is weird, here is a better implementation:
```python
image_path.rsplit("_", 1)[0]
```